### PR TITLE
Add GA event tracking for the No Audiences Banner and Info Notice

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #9496 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding sxandards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/audiences.js [hunk 1]
```diff
@@ -637,15 +637,19 @@ const baseActions = {
...
-		// Expire new badges for initially configured audiences.
-		yield commonActions.await(
-			dispatch( CORE_USER ).setExpirableItemTimers(
-				configuredAudiences.map( ( slug ) => ( {
-					slug: `${ AUDIENCE_ITEM_NEW_BADGE_SLUG_PREFIX }${ slug }`,
-					expiresInSeconds: 1,
-				} ) )
-			)
-		);
+		if ( configuredAudiences.length > 0 ) {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/audiences.js:640]: This fixes a minor bug, calling `setExpirableItemTimers()` with an empty list results in an error being returned and logged to the console.

File: assets/js/modules/analytics-4/datastore/audiences.js [hunk 2]
```diff
@@ -637,15 +637,19 @@ const baseActions = {
...
+				dispatch( CORE_USER ).setExpirableItemTimers(
+					configuredAudiences.map( ( slug ) => ( {
+						slug: `${ AUDIENCE_ITEM_NEW_BADGE_SLUG_PREFIX }${ slug }`,
+						expiresInSeconds: 1,
+					} ) )
+				)
+			);
+		}
+
+		return {};
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/audiences.js:652]: This fixes an error that occurs when attempting to destructure the return value of the action.

File: assets/js/modules/analytics-4/datastore/audiences.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/datastore/audiences.js:640]: Nice one. Thanks for fixing it, @techanvil 👍

**Review by hussain-t — APPROVED** [review]
Excellent work, @techanvil! LGTM 👍

**Comment by hussain-t** [comment]
Note: I've verified that the CI build failures are unrelated to this PR.

**Pull request merged**